Make aoe_ipauth compatible with TYPO3 11

christianbltr opened the pull request:
This makes aoe_ipauth compatible with TYPO3 11.

Tested with TYPO3 11.5.27.

Only the "Auto-Login User" feature has been tested, not the "Auto-Login Group" feature.

see https://github.com/AOEpeople/aoe_ipauth/issues/61

Commit:
[TASK] Make groups available immediately after login, fixes https://github.com/AOEpeople/aoe_ipauth/issues/60

Commit:
[TASK] Add hook which will be fired after user has been authenticated

Commit:
[TASK] Make extension installable with TYPO3 11

Commit:
[TASK] Set extension key

Commit:
[TASK] Remove TYPO3_MODE and $_EXTKEY

Commit:
[TASK] Apply changes proposed by rector

Commit:
[TASK] Add icon

derhansen committed:
[!!!][BUGFIX] Fix fe_groups lookup by ip address

This change fixes the fe_groups lookup by ip address.
It however also removes compatibility to TYPO3 9.5 and
10.4, since the `ModifyResolvedFrontendGroupsEvent` is
only available in TYPO3 11.5

christianbltr committed:
Merge pull request #1 from derhansen/task/fe-group-auth

[!!!][BUGFIX] Fix fe_groups lookup by ip address

christianbltr commented:
This merge request now also includes this patch from @derhansen :

> [!!!][BUGFIX] Fix fe_groups lookup by ip address
>This change fixes the fe_groups lookup by ip address.
> It however also removes compatibility to TYPO3 9.5 and
> 10.4, since the `ModifyResolvedFrontendGroupsEvent` is
> only available in TYPO3 11.5

Would you consider merging it and releasing a version for TYPO3 11?

tlayh commented:
We do not further develop this extension because we currently don't use it.

Feel free to update the pull request, so that at least the actions (Tests) are green again and then I can merge it for you. For me it would be enough if the tests for TYPO3 10.4 and 11.5 are green. PHP Version can be above 7.4.

christianbltr commented:
@tlayh Are you still willing to merge this? I cannot see any actions anymore, maybe they have been removed?

We will also migrate the extension to TYPO3 12. Are you interested in merging that also?

Commit:
[TASK] Update dependencies for TYPO3 v12 compatibility

Update `typo3/cms-core` to support version 12. Replace deprecated
development dependencies with updated packages. Add composer plugin
allow-plugins configuration for compatibility with TYPO3’s requirements.

Commit:
[FEATURE] Upgrade to TYPO3 v12 and refactor IP auth extension

Upgrade the extension's compatibility to TYPO3 v12
and adjust codebase for deprecated API replacements.
Remove ext_tables.php, enforce strong typing, and
generalize relevant utilities using GeneralUtility.
